src/data-workspace/category-combo-table/category-combo-table.js

```diff
@@ -8,55 +8,63 @@ import {
+import { useMetadata, selectors } from '../../metadata/index.js'
+import { cartesian } from '../../shared/utils.js'
 import {
     DataEntryCell,
     DataEntryField,
     useActiveCell,
-} from './data-entry-cell/index.js'
+} from '../data-entry-cell/index.js'
+import styles from './category-combo-table.module.css'
+import { TotalHeader, ColumnTotals, RowTotal } from './total-cells.js'
 
 export const CategoryComboTable = ({
     categoryCombo,
     dataElements,
     filterText,
     globalFilterText,
     maxColumnsInSection,
+    renderRowTotals,
+    renderColumnTotals,
 }) => {
     const { data } = useMetadata()
     const { deId: activeDeId, cocId: activeCocId } = useActiveCell()
 
-    if (!data) {
-        return null
-    }
-
     const categories = selectors.getCategoriesByCategoryComboId(
         data,
         categoryCombo.id
     )
 
-    // each element is a combination of category-options for a particular column
-    // this results in lists of category-options in the same order as headers are rendered
-    // the result is a client side computation of categoryOption-combinations
-    const optionsIdLists = categories.map((cat) => cat.categoryOptions)
-    const computedCategoryOptions = cartesian(optionsIdLists)
+    const sortedCOCs = useMemo(() => {
+        // each element is a combination of category-options for a particular column
+        // this results in lists of category-options in the same order as headers are rendered
+        // the result is a client side computation of categoryOption-combinations
+        const optionsIdLists = categories.map((cat) => cat.categoryOptions)
+        const computedCategoryOptions = cartesian(optionsIdLists)
+        // find categoryOptionCombos by category-options
+        return computedCategoryOptions
+            .map((options) =>
+                selectors.getCoCByCategoryOptions(
+                    data,
+                    categoryCombo.id,
+                    options
+                )
+            )
+            .filter((coc) => !!coc)
+    }, [data, categories, categoryCombo])
 
-    if (
-        computedCategoryOptions.length !==
-        categoryCombo.categoryOptionCombos?.length
-    ) {
+    if (sortedCOCs.length !== categoryCombo.categoryOptionCombos?.length) {
         console.warn(
             `Computed combination of categoryOptions for catCombo(${categoryCombo.id}) is different from server. 
             Please regenerate categoryOptionCombos. 
-            Computed: ${computedCategoryOptions.length}
+            Computed: ${sortedCOCs.length}
             Server: ${categoryCombo.categoryOptionCombos.length})`
         )
     }
     // Computes the span and repeats for each columns in a category-row.
     // Repeats are the number of times the options in a category needs to be rendered per category-row
-    let catColSpan = computedCategoryOptions.length
+    let catColSpan = sortedCOCs.length
     const rowToColumnsMap = categories.map((c) => {
```

Comment:
Yeah I've thought about refactoring this to a header component, but thought maybe this would be best in it's own PR, since it's not really relevant to this feature. The reason I started some refactoring was because I implemented this feature differently and thus touched upon this code, but figured it was not needed now.

There are comments here about what it does though. I guess it would be more clear in a separate component/function. This basically just computes the headers in the table, which is a bit complex due to the dynamic nature of the table, since you need to calculate the columnSpan and number of times each option should be repeated. This is explained in the comments, and I'm not sure this would be much clearer with more comments to be honest, but I can try in a refactor. 

The reason `columnSpan` is overridden is because it depends on the previous. Of course you could probably write this using a `previousColumnSpan` and calculate `currentColumnSpan` from this, but the end result is the same (with more lines). Would that make it clearer?

 We are creating a "nested"/pivot-table, and for it to make sense, all the options in the previous category need to "span" all options in the current category. `sortedCOCs.length` will refer to the number of rendered columns, and this is used as the base, because in the end the last category should have the same amount of columns - and reach span "1". But to get there, we are basically doing factorization in each step, and the result in each step is the span for that category.

Eg. table with one category is simple, each column has a span of 1, so this loop only happens once. Say there are 4 options; `sortedCOCs.length === 4, nrOfOptions ===4` `4/4 === 1`, span is 1. 

Eg. table with 2 categories. 
`Category1 : Location:     <Fixed, Outreach>`
`Category2: Age:   <1, >1.`
sortedCOCS.length === 4. 
`1. iteration (Location):
     catColSpan = 4/numberOfOptions === 4/2 === 2
`
`2. iteration (Age)
   catColSpan = 2/numberOfOptions === 2/2 === 1
`
You see that current columnSpan depends on previous. There are probably dozens of way to write this as well, and as mentioned maybe a `previousColumnSpan` is clearer?
I think that in situations like this, if you really want to understand the code you need to take some time to study the problem anyway.

Again, this might be more easier to follow in a `CategoryComboTableHeader`-component. But I do think we should keep that refactor out of this PR.

Reply:
I'm also having a tough time completely following this logic. Quite likely that's because I am unfamiliar with some intricacies of what we are trying to do here. In any case, I think that doing some refactoring here would be very beneficial and we can look into simplifying things when we do that. I fully agree that it should be done in a subsequent PR.